usl_pipeline/usl_lib/usl_lib/chunkers/elevation_chunkers.py

```diff
@@ -0,0 +1,46 @@
+from osgeo import gdal
+import typing
+
+from usl_lib.readers import elevation_readers
+from usl_lib.shared import suppliers
+
+
+def split_geotiff_into_chunks(
+    elevation_file_path: str,
+    chunk_size: int,
+    chunk_file_path_generator: suppliers.ChunkFilePathGenerator,
+) -> typing.Tuple[int, int]:
+    """Produces a grid of chunk GeoTIFF files based on input GeoTIFF file.
+
+    Args:
+        elevation_file_path: GeoTIFF file with elevation data to be split into chunks.
+        chunk_size: Size of a chunk in cells (along both X and Y axes).
+        chunk_file_path_generator: Caller provided source to generate chunk file names.
+
+    Returns:
+        Tuple (number of chunks along the Y-axis, number of chunks along the X-axis).
```

Comment:
I think it'd be even more helpful to return the file paths for the chunks. We'd like to be able to write something like
```python
for chunk in split_into_chunks():
  upload_to_gcp(chunk)
```
so the caller doesn't have to recreate the path names.

Reply:
In order to avoid having the same names for different types of inputs (elevation, buildings and so on) I added optional file pattern parameter and also each chunk is now represented by not just string or path but more like a descriptor with y- and x- index in it.